### PR TITLE
Document `pick_random` for empty arrays

### PR DESCRIPTION
**Pull request by novalis**

### CODE REVIEW

**Comment by Sauermann** [comment]
Executing `pick_random` on an empty array results in the following error message:
```
E 0:00:02:0415   Control.gd:64 @ _on_button_2_pressed(): Can't take value from empty array.
  <C++ Error>    Condition "_p->array.size() == 0" is true. Returning: Variant()
  <C++ Source>   core/variant/array.cpp:329 @ pick_random()
  <Stack Trace>  Control.gd:64 @ _on_button_2_pressed()
```
So I'm not entirely convinced, that documenting a behavior which causes an error message is warranted.

**Comment by AThousandShips** [comment]
We can possibly document that it requires a non empty array, though that *should* be obvious

**Comment by Sauermann** [comment]
Sounds good. The documentation of `Array.back` and `Array.front` could be used for that.

File: doc/classes/Array.xml [hunk 1]
```diff
@@ -462,7 +462,7 @@
 		<method name="pick_random" qualifiers="const">
 			<return type="Variant" />
 			<description>
-				Returns a random value from the target array.
+				Returns a random value from the target array. If the array is empty, returns [code]null[/code].
```
Comment (AThousandShips) [inline doc/classes/Array.xml:465]:
```suggestion
				Returns a random value from the target array. Prints an error and returns [code]null[/code] if the array is empty.
```

**Review by AThousandShips — APPROVED** [review]
LGTM, clear and concise

**Comment by akien-mga** [comment]
Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**